bundles/org.openhab.binding.heos/src/main/java/org/openhab/binding/heos/internal/api/HeosSystem.java

```diff
@@ -45,157 +41,67 @@
-            return false;
-        }
+    private final HeosJsonParser parser = new HeosJsonParser();
+
+    private ScheduledExecutorService keepAliveExecutor;
+
+    public HeosResponseObject<Void> send(String command) {
+        return send(command, Void.class);
     }
 
-    /**
-     * This method returns a class from type HeosCommands. Within this class all
-     * possible commands to control the HEOS network are defined.
-     *
-     * @return a class with all HEOS Commands
-     */
-    public HeosCommands command() {
-        return heosCommand;
+    // FIXME reconsider throwing exceptions here?
+    public <T> HeosResponseObject<T> send(String command, Class<T> clazz) {
+        try {
+            return sendCommand.send(command, clazz);
+        } catch (ReadException e) {
+            logger.debug("HEOS System read failure during response. message: {}", e.getMessage());
+            logger.debug("HEOS failed command: {}", command);
+        } catch (IOException e) {
+            logger.debug("IO Exception during send HEOS command with message: {}", e.getMessage());
+        }
+        return null;
     }
 
     /**
      * Establishes the connection to the HEOS-Network if IP and Port is
      * set. The caller has to handle the retry to establish the connection
      * if the method returns {@code false}.
      *
-     * @param connectionDelay if set to true system waits for 10 seconds before connecting to event line
+     * @param connectionIP
+     * @param connectionPort
      * @return {@code true} if connection is established else returns {@code false}
      */
-    public boolean establishConnection(boolean connectionDelay) {
-        this.commandLine = new Telnet();
-        this.eventLine = new Telnet();
-
-        boolean commandLineConnected = false;
-        boolean eventLineConnected = false;
-
-        try {
-            commandLineConnected = commandLine.connect(connectionIP, connectionPort);
-        } catch (IOException e) {
-            logger.debug("Exception during connection to bridge with message: {}", e.getMessage());
-//            updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.COMMUNICATION_ERROR, e.getMessage());
-            logger.debug("Could not connect HEOS command line at IP {} @ port {}", connectionIP, connectionPort);
-            commandLineConnected = false;
-        }
-        if (commandLineConnected) {
+    public boolean establishConnection(String connectionIP, int connectionPort) throws IOException, ReadException {
+        if (commandLine.connect(connectionIP, connectionPort)) {
             logger.debug("HEOS command line connected at IP {} @ port {}", connectionIP, connectionPort);
-            sendCommand.setTelnetClient(commandLine);
-            send(command().registerChangeEventOFF());
+            // TODO does it make sense to send this to the normal sendcommand endpoint?
+            // send(sendCommand, command().registerChangeEventOFF());
```

Comment:
Hi Martin,
if I remember correctly I did this to make sure that the HEOS system does not start to receive ChangeEvents after a reconnection on this line. At the beginning of a connection the HEOS system send a lot of waste after the connection has established. But maybe this is better now after a lot of firmware updates.